### ISSUE TITLE
Pattern Matching for Path and Filename

### ISSUE THREAD

**Issue by SeriousM** [Ragnoroct/linkme#9]
Hey, I was about to build such an extension myself as I use tags in my todo-files to reference devops tickets in format of `@#1234`.
Because we have different projects the `1234` could mean project A or B or C.
I would like to create a rule for a specific filename or extension in specific paths.
That means separate regex matching for filename and path.

What do you think, is this hard to implement?

**Comment by arkadianriver** [comment]
Hi! I just discovered this extension, too, and love it. @SeriousM I don't quite follow the second part about filenames and extensions, but for the project A,B,C question, if you're referring to _multiple_ replaceable parameters, it can do that, too! .. and JS seems to even match utf8 characters!

```
"linkme.rules": [
  {
    "pattern": "@#(.*)",
    "url": "https://github.com/yourdefault/repo/issues/\\1"
  },
  {
    "pattern": "@•(.*)#(.*)",
    "url": "https://github.com/\\1/issues/\\2"
  }
]
```
**Note:** tag modified slightly assuming you might be talking about the Todo+ extension? If so, to not match all tags, I added the `•` character to denote a github-issue-specific tag, like the `#` pound sign does. Also using a different character than the pound sign so that it doesn't double match both patterns. Those would match entries like this `@#112` and `@•ragnoroct/linkme#2`.

Is this what you're looking for or did I misunderstand?

**Comment by arkadianriver** [comment]
Oh wait, now I get what you're asking for. Something like this right?
```
    "linkme.rules": {
        "default": [
            ...patterns...
        ],
        "projA*.todo": [
            ...patterns...
        ],
        "projB*.todo":  [
            ...patterns...
        ]
    }
```

**Comment by SeriousM** [comment]
Yes that looks like it, just that the name pattern may contain a path as well, it it's not too complicated.
